webextension/background.js

```diff
@@ -412,17 +423,36 @@ const TabState = (function() {
       return this._report.screenshot;
     }
 
-    updateReport(data) {
-      if (Object.keys(data).length) {
-        for (const [name, value] of Object.entries(data)) {
+    updateReport(updates) {
+      if (updates === undefined) {
+        this._report = {};
+      } else {
+        for (const [name, value] of Object.entries(updates)) {
           if (value === undefined) {
             delete this._report[name];
           } else {
             this._report[name] = value;
           }
         }
-      } else {
-        this._report = {};
+      }
+
+      if (!this._report.appVersion) {
```

Comment:
Is there a circumstance where we're concerned about overwriting one of these props on `this._report` ?

Reply:
Probably not, but I was just in "defensive coding" mode, and this is the defensive strategy it snapped to :)  I'm not against dropping the `if` guards, if you'd like.

Reply:
Cool, no need to change it.